worker/src/queues/cloudUsageMeteringQueue.ts

```diff
@@ -1,36 +1,60 @@
 import { Processor, Queue } from "bullmq";
 import {
-  redis,
   logger,
   QueueName,
   QueueJobs,
+  createNewRedisInstance,
 } from "@langfuse/shared/src/server";
 import { handleCloudUsageMeteringJob } from "../ee/cloudUsageMetering/handleCloudUsageMeteringJob";
 import { env } from "../env";
 
-export const cloudUsageMeteringQueue =
-  redis && env.STRIPE_SECRET_KEY
-    ? new Queue(QueueName.CloudUsageMeteringQueue, {
-        connection: redis,
-      })
-    : null;
-
-if (cloudUsageMeteringQueue) {
-  cloudUsageMeteringQueue.add(
-    QueueJobs.CloudUsageMeteringJob,
-    {},
-    {
-      repeat: { pattern: "5 * * * *" },
-    },
-  );
-
-  // add a job to the queue to start the job immediately in case we need to catch up
-  cloudUsageMeteringQueue.add(QueueJobs.CloudUsageMeteringJob, {}, {});
-
-  // log the jobs in the queue
-  cloudUsageMeteringQueue.getJobCounts().then((counts) => {
-    logger.info("Cloud Usage Metering Queue", counts);
-  });
+export class CloudUsageMeteringQueue {
+  private static instance: Queue | null = null;
+
+  public static getInstance(): Queue | null {
+    if (!env.STRIPE_SECRET_KEY) {
+      return null;
+    }
+
+    if (CloudUsageMeteringQueue.instance) {
+      return CloudUsageMeteringQueue.instance;
+    }
+
+    const newRedis = createNewRedisInstance({ enableOfflineQueue: false });
+
+    CloudUsageMeteringQueue.instance = newRedis
+      ? new Queue(QueueName.CloudUsageMeteringQueue, {
+          connection: newRedis,
+          defaultJobOptions: {
+            removeOnComplete: true,
+            removeOnFail: 100_000,
```

Comment:
*question*: Do we need to keep that many failed jobs? Job is running on repeat, so whether an individual job has failed should not matter that much

Reply:
Good point, I can reduce that to a couple 100.